src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -1744,9 +1742,9 @@ void ShenandoahHeap::op_final_mark() {
       set_has_forwarded_objects(true);
 
       if (!is_degenerated_gc_in_progress()) {
-        if (ShenandoahConcurrentRoots::should_do_concurrent_class_unloading()) {
-          ShenandoahCodeRoots::arm_nmethods();
-        }
+        // We need to arm nmethods for concurrent codecache process, regardless concurrent
+        // class unloading
```

Comment:
Comment should probably not refer to previous implementations. Future readers will not see the connection there.

Reply:
Thanks for reviewing.

Comment updated to "Arm nmethods for concurrent codecache processing."